pkg/upgrade/upgrades/permanent_maybe_setup_pcr_reader.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/replication"
+	"github.com/cockroachdb/cockroach/pkg/upgrade"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+)
+
+func maybeSetupPCRStandbyReader(
```

Comment:
Not quite. This upgrade is run by the tenant when it starts. It isn't started until it has service enabled, which is done by the pcr job after the scan is finished.

Reply:
That said, I agree with the suggestion that we should avoid calling `SetupOrAdvanceStandbyReaderCatalog` if we see a zero ts, so in the poller job we'd `continue` if it is zero, and in this bootstrap upgrade we'd just log a warning and proceed to only setup the poller job if it was zero.

But I'd still keep the call in the bootstrap, since we run bootstrap before opening up to SQL queries, just to try to avoid having those queries come in before the catalog is setup and race with the poller job.